maint-lib/makelib.mk

```diff
@@ -19,11 +19,11 @@ $(shell bash -c 'set -eu ; \
 	set_var CEPH_POINT_RELEASE "$$(bash maint-lib/ceph_version.sh "$$ceph_version_spec" CEPH_POINT_RELEASE)" ; \
 	set_var BASEOS_REPO        "$(word 3, $(subst $(comma), , $(1)))" ; \
 	set_var BASEOS_TAG         "$(word 4, $(subst $(comma), , $(1)))" ; \
-	set_var BASEOS_REGISTRY    "_" ; \
+	set_var BASEOS_REGISTRY    "$(BASEOS_REGISTRY)" ; \
 	set_var IMAGES_TO_BUILD    "$(IMAGES_TO_BUILD)" ; \
 
 	set_var STAGING_DIR       "staging/$$CEPH_VERSION$$CEPH_POINT_RELEASE-$$BASEOS_REPO-$$BASEOS_TAG-$$HOST_ARCH" ; \
-	base_img="$$BASEOS_REGISTRY/$$BASEOS_REPO:$$BASEOS_TAG" ; \
+	base_img="$$BASEOS_REPO:$$BASEOS_TAG" ; \
```

Comment:
I see a couple problems with this change. I think this code will break builds. 

-----

Something that I missed is that in `src/daemon-base/Dockerfile`, we have 
```
FROM __ENV_[BASE_IMAGE]__
```
so we need the `BASE_IMAGE` variable to have the full <registry (unless '_')>/<repo>:<tag>.

I think the best thing to do is change the name of the `BASE_IMAGE` variable in the source to `BASE_IMAGE_FULLPATH` or something similar.

-----

For the second, this change doesn't ever propagate a user-specified `BASEOS_REGISTRY` forward. As a variable, it is never used here to set what is currently the `BASE_IMAGE` variable (and what I've proposed will become the `BASE_IMAGE_FULLPATH` variable). The line part `"$${base_img#_/}"` below is what strips off the leading `_/` if the registry is left as default (and leaves the registry intact if it is any other string).

Reply:
It occurs to me also that `BASE_IMAGE` should be a required env var in staging, but it isn't set as required. I'll work on a patch to fix this, and I'll plan to rebase it after we merge this.